### ISSUE TITLE
Make runtest issue

### ISSUE THREAD

**Issue by gangooteli** [rbgirshick/py-faster-rcnn#761]
From last 2 weeks I am trying to setup caffe.
This is frustating. Some how using docker i was able to run following commands
make all & make test

Now stuck in make runtest
Getting below error:

.build_release/tools/caffe
caffe: command line brew
usage: caffe <command> <args>

commands:
  train           train or finetune a model
  test            score a model
  device_query    show GPU diagnostic information
  time            benchmark model execution time

  Flags from tools/caffe.cpp:
    -gpu (Optional; run in GPU mode on given device IDs separated by ','.Use
      '-gpu all' to run on all available GPUs. The effective training batch
      size is multiplied by the number of devices.) type: string default: ""
    -iterations (The number of iterations to run.) type: int32 default: 50
    -level (Optional; network level.) type: int32 default: 0
    -model (The model definition protocol buffer text file.) type: string
      default: ""
    -phase (Optional; network phase (TRAIN or TEST). Only used for 'time'.)
      type: string default: ""
    -sighup_effect (Optional; action to take when a SIGHUP signal is received:
      snapshot, stop or none.) type: string default: "snapshot"
    -sigint_effect (Optional; action to take when a SIGINT signal is received:
      snapshot, stop or none.) type: string default: "stop"
    -snapshot (Optional; the snapshot solver state to resume training.)
      type: string default: ""
    -solver (The solver definition protocol buffer text file.) type: string
      default: ""
    -stage (Optional; network stages (not to be confused with phase), separated
      by ','.) type: string default: ""
    -weights (Optional; the pretrained weights to initialize finetuning,
      separated by ','. Cannot be set simultaneously with snapshot.)
      type: string default: ""
.build_release/test/test_all.testbin 0 --gtest_shuffle 
Cuda number of devices: 0
Setting to use device 0
Current device id: 0
Current device name: 
Note: Randomizing tests' orders with a seed of 80236 .
[==========] Running 2123 tests from 281 test cases.
[----------] Global test environment set-up.
[----------] 26 tests from NetTest/2, where TypeParam = caffe::GPUDevice<float>
[ RUN      ] NetTest/2.TestHasLayer
E0105 18:15:34.966894  4654 common.cpp:114] Cannot create Cublas handle. Cublas won't be available.
E0105 18:15:34.986889  4654 common.cpp:121] Cannot create Curand generator. Curand won't be available.
F0105 18:15:34.990039  4654 syncedmem.hpp:22] Check failed: error == cudaSuccess (30 vs. 0)  unknown error
*** Check failure stack trace: ***
    @     0x7f9e0ea855cd  google::LogMessage::Fail()
    @     0x7f9e0ea87433  google::LogMessage::SendToLog()
    @     0x7f9e0ea8515b  google::LogMessage::Flush()
    @     0x7f9e0ea87e1e  google::LogMessageFatal::~LogMessageFatal()
    @     0x7f9e0c9e1f71  caffe::SyncedMemory::mutable_cpu_data()
    @     0x7f9e0ca06a70  caffe::Blob<>::Reshape()
    @     0x7f9e0ca070e4  caffe::Blob<>::Reshape()
    @     0x7f9e0c951178  caffe::DummyDataLayer<>::LayerSetUp()
    @     0x7f9e0c9f7077  caffe::Net<>::Init()
    @     0x7f9e0c9f97be  caffe::Net<>::Net()
    @           0x4f88fc  caffe::NetTest<>::InitNetFromProtoString()
    @           0x4a150b  caffe::NetTest<>::InitTinyNet()
    @           0x4ad3fe  caffe::NetTest_TestHasLayer_Test<>::TestBody()
    @           0x92c2b3  testing::internal::HandleExceptionsInMethodIfSupported<>()
    @           0x9258ca  testing::Test::Run()
    @           0x925a18  testing::TestInfo::Run()
    @           0x925af5  testing::TestCase::Run()
    @           0x926dcf  testing::internal::UnitTestImpl::RunAllTests()
    @           0x9270f3  testing::UnitTest::Run()
    @           0x46df9d  main
    @     0x7f9e0bbd5830  __libc_start_main
    @           0x475af9  _start
    @              (nil)  (unknown)
Makefile:532: recipe for target 'runtest' failed
make: *** [runtest] Aborted (core dumped)

I am totally exhausted by whole setup process doing multiple times for last 2 weeks
